edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/storage/MessageStoreTest.cs

```diff
@@ -297,6 +297,88 @@ public async Task CleanupTestTimeoutUpdateTimeToLive()
+                        IMessage updatedMessage = await messageStore.Add("module1", input, 20);
+                        CompareUpdatedMessageWithOffset(input, i / 2, updatedMessage);
+                    }
+                    else
+                    {
+                        IMessage input = this.GetMessage(i);
+                        IMessage updatedMessage = await messageStore.Add("module2", input, 20);
+                        CompareUpdatedMessageWithOffset(input, i / 2, updatedMessage);
+                    }
+                }
+
+                IMessageIterator module1Iterator = messageStore.GetMessageIterator("module1");
+                IEnumerable<IMessage> batch = await module1Iterator.GetNext(100);
+                Assert.Equal(100, batch.Count());
+
+                IMessageIterator module2Iterator = messageStore.GetMessageIterator("module2");
+                batch = await module2Iterator.GetNext(100);
+                Assert.Equal(100, batch.Count());
+
+                await Task.Delay(TimeSpan.FromSeconds(100));
+
+                module1Iterator = messageStore.GetMessageIterator("module1");
+                batch = await module1Iterator.GetNext(100);
+                Assert.Empty(batch);
+
+                module2Iterator = messageStore.GetMessageIterator("module2");
+                batch = await module2Iterator.GetNext(100);
+                Assert.Empty(batch);
+
+                // By setting the global TTL for the MessageStore to 20, the CleanupProcessor will run every 10 seconds
+                // But it won't clean up any messages, since the individual messages are set to have TTL of 2000 seconds
+                result.messageStore.SetTimeToLive(TimeSpan.FromSeconds(20));
+                await Task.Delay(TimeSpan.FromSeconds(50));
+
+                for (int i = 0; i < 200; i++)
+                {
+                    if (i % 2 == 0)
+                    {
+                        IMessage input = this.GetMessage(i);
+                        IMessage updatedMessage = await messageStore.Add("module1", input, 2000);
+                        CompareUpdatedMessageWithOffset(input, 100 + i / 2, updatedMessage);
+                    }
+                    else
+                    {
+                        IMessage input = this.GetMessage(i);
+                        IMessage updatedMessage = await messageStore.Add("module2", input, 2000);
+                        CompareUpdatedMessageWithOffset(input, 100 + i / 2, updatedMessage);
+                    }
+                }
+
+                module1Iterator = messageStore.GetMessageIterator("module1");
+                batch = await module1Iterator.GetNext(100);
+                Assert.Equal(100, batch.Count());
+
+                module2Iterator = messageStore.GetMessageIterator("module2");
+                batch = await module2Iterator.GetNext(100);
+                Assert.Equal(100, batch.Count());
+
+                await Task.Delay(TimeSpan.FromSeconds(100));
```

Comment:
>await Task.Delay(TimeSpan.FromSeconds(100)); [](start = 16, length = 44)

prob worthwhile to make module1 and module2 messages have different TTL, then wait out and difference and see that only one set of messages get iterated.  This will likely be the most frequent scenario.

Reply:
Makes sense. I've made that change